core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -1165,6 +1165,46 @@ protected ExprEval eval(double param)
     }
   }
 
+  class SafeDivide extends BivariateMathFunction
+  {
+    public static final String NAME = "safe_divide";
+
+    @Override
+    public String name()
+    {
+      return NAME ;
+    }
+
+    @Nullable
+    @Override
+    public ExpressionType getOutputType(Expr.InputBindingInspector inspector, List<Expr> args)
+    {
+      return ExpressionTypeConversion.integerMathFunction(
+          args.get(0).getOutputType(inspector),
+          args.get(1).getOutputType(inspector)
+      );
+    }
+
+    @Override
+    protected ExprEval eval(final long x, final long y)
+    {
+      if(y==0) {
+        return ExprEval.of(0);
+      }
```

Comment:
hmm, looking at the behavior of another `SAFE_DIVIDE` I could find in other databases (https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#safe_divide), I think we should consider returning `null` here if `NullHandling.replaceWithDefault()` is true

```suggestion
      if (y == 0 && x != 0) {
        return ExprEval.ofLong(NullHandling.defaultLongValue());
      }
```

Reply:
Change made. Added a case and testcases where x or y is NaN or +- Infinity